modules/features2d/src/sift.dispatch.cpp

```diff
@@ -117,13 +117,17 @@ class SIFT_Impl : public SIFT
     CV_PROP_RW double contrastThreshold;
     CV_PROP_RW double edgeThreshold;
     CV_PROP_RW double sigma;
+    CV_PROP_RW bool useUcharDescriptors;
 };
 
 Ptr<SIFT> SIFT::create( int _nfeatures, int _nOctaveLayers,
-                     double _contrastThreshold, double _edgeThreshold, double _sigma )
+                     double _contrastThreshold, double _edgeThreshold, double _sigma, int _descriptorType )
 {
     CV_TRACE_FUNCTION();
-    return makePtr<SIFT_Impl>(_nfeatures, _nOctaveLayers, _contrastThreshold, _edgeThreshold, _sigma);
+
+    CV_Assert(_descriptorType == CV_32F || _descriptorType == CV_8U);
+    bool useUcharDescriptors = (_descriptorType == CV_8U);
```

Comment:
IMO it make sense to propagate the type info deeper to `SIFT_Impl` itself to simplify `descriptorType()` and allocation of `_descriptors`. I also suggest providing a comment why `_descriptorType` is restricted to 8U and 32F since it's not obvious here.